### PR TITLE
Virtualizer logic integration

### PR DESCRIPTION
**Pull request by wyf-1116**
To be reviewed

### CODE REVIEW

**Comment by ruxandraRosu** [comment]
recheck

**Comment by planetf1** [comment]
Note builds are currently broken. see https://jira.linuxfoundation.org/servicedesk/customer/portal/2/IT-16387

**Comment by askb** [comment]
recheck

File: open-metadata-test/open-metadata-fvt/open-types-fvt/open-types-test/src/main/java/org/odpi/openmetadata/fvt/opentypes/entities/InformationView/InformationViewReferences.java [hunk 1]
```diff
@@ -0,0 +1,989 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+
+// This is a generated file - do not edit - changes should be made to the templates amd/or generator to generate this file with changes.
```
Comment (bogdan-sava) [inline open-metadata-test/open-metadata-fvt/open-types-fvt/open-types-test/src/main/java/org/odpi/openmetadata/fvt/opentypes/entities/InformationView/InformationViewReferences.java]: It says this is generated file.
Does it means suppose NOT to be committed.

File: open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-server/src/main/java/org/odpi/openmetadata/governanceservers/virtualizationservices/admin/VirtualizationOperationalServices.java [hunk 2]
```diff
@@ -166,10 +194,22 @@ public void initialize(VirtualizationConfig virtualizationConfig, OMRSAuditLog a
                 }
             }
 
+            if (virtualizationSolutionConnector != null) {
+                try {
+                    virtualizationSolutionConnector.start();
+                    /*
```
Comment (bogdan-sava) [inline open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-server/src/main/java/org/odpi/openmetadata/governanceservers/virtualizationservices/admin/VirtualizationOperationalServices.java]: please remove this comment

File: open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-api/src/main/java/org/odpi/openmetadata/governanceservers/virtualizationservices/event/VirtualizerTopicListener.java [hunk 3]
```diff
@@ -33,11 +55,101 @@ public VirtualizerTopicListener(OpenMetadataTopicConnector ivInTopicConnector){
...
+                log.info("No views were created in Gaian, nothing to publish");
+            } else {
+                for (Map.Entry<String, String> entry : createdViews.entrySet()) {
+                    if (entry.getValue() == null) {
+                        log.info("No view was created for " + entry.getKey());
+                    } else {
+                        events.add(createInformationViewEvent(entry.getKey(), entry.getValue(), tableContextEvent));
+                    }
+                }
+                //sendNotification(events);
```
Comment (bogdan-sava) [inline open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-api/src/main/java/org/odpi/openmetadata/governanceservers/virtualizationservices/event/VirtualizerTopicListener.java]: log.debug message doesn't seems to match with the code.
And please remove commented code

File: open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-api/pom.xml [hunk 4]
```diff
@@ -14,20 +14,38 @@
...
     </parent>
 
-    <name>Virtualization Services API</name>
-    <description>
-        The common objects and function sed both in the client and the server.
-    </description>
-
     <artifactId>virtualization-services-api</artifactId>
-
+    <build>
```
Comment (bogdan-sava) [inline open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-api/pom.xml]: maven-compiler-plugin configuration is not needed here

File: open-metadata-test/open-metadata-fvt/open-types-fvt/open-types-test/src/main/java/org/odpi/openmetadata/fvt/opentypes/entities/InformationView/InformationViewReferences.java (hunk 1 above)
Comment (planetf1) [reply open-metadata-test/open-metadata-fvt/open-types-fvt/open-types-test/src/main/java/org/odpi/openmetadata/fvt/opentypes/entities/InformationView/InformationViewReferences.java]: Even more importantly... how is this file now in the proposed commit given that it was deleted a while ago?
Have any force-push been used (which could contaminate history?)
Could this cause any other odd changes to be pulled in?

Comment (planetf1) [reply open-metadata-test/open-metadata-fvt/open-types-fvt/open-types-test/src/main/java/org/odpi/openmetadata/fvt/opentypes/entities/InformationView/InformationViewReferences.java]: I can see the commit #51c9ab4e9c180cf6226250c1b8f07d76ba990176 is also in this branch, so it's
likely this file was accidentally re-added, before the later fix to remove from .gitignore

Please remove from the commit.

Comment (bogdan-sava) [reply open-metadata-test/open-metadata-fvt/open-types-fvt/open-types-test/src/main/java/org/odpi/openmetadata/fvt/opentypes/entities/InformationView/InformationViewReferences.java]: No force push and will not be used without agreement

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java [hunk 5]
```diff
@@ -0,0 +1,463 @@
...
+
+        final String actionDescription = "executeCustomizedQuery: " + query;
+
+        try {
+            derbyStatement = derbyConnection.createStatement();
+            derbyStatement.setQueryTimeout(timeoutInSecond);
+            derbyStatement.executeQuery(query);
+            return true;
+        } catch (SQLException e) {
+            log.error("Error deleting table", e);
```
Comment (ruxandraRosu) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java]: I would be useful to have a meaningful error message, the query here is not only for delete

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java [hunk 6]
```diff
@@ -0,0 +1,463 @@
...
+                        auditCode.getUserAction());
+            }
+        }
+        return logicTableList;
+    }
+
+
+    public boolean executeCustomizedQuery(String query) {
+
+        final String actionDescription = "executeCustomizedQuery: " + query;
```
Comment (ruxandraRosu) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java]: Query could be logged on debug .level

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java [hunk 7]
```diff
@@ -0,0 +1,463 @@
...
+     * @param logicalTableName
+     * @param gaianNodeName
+     */
+    private void createMirroringLogicalTable(String logicalTableName, String gaianNodeName) {
+        log.debug("Set up Logical Table for Gaian node");
+        String setLogicalTableForNode = "call setltfornode('" +
+                logicalTableName + "','" +
+                gaianNodeName + "')";
+
+        executeCustomizedQuery(setLogicalTableForNode);
```
Comment (ruxandraRosu) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java]: A new method calling executeUpdate should be used instead of executeQuery as this statement is not a query, it doesn't return any results.

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java [hunk 8]
```diff
@@ -0,0 +1,463 @@
...
+    }
+
+    private String createTableDefinition(String databaseName, String tableName, Function<MappedColumn, String> function, List<MappedColumn> mappedColumns, String gaianNodeName, String logicalTableName) {
+        String businessTableCreateStatement = buildTableCreateStatement(tableName, mappedColumns, function);
+        String setBusinessTableDataSource = buildCreateTableDataSourceStatement(databaseName, tableName, gaianNodeName, mappedColumns, logicalTableName);
+
+        boolean queryStatus = executeCustomizedQuery(businessTableCreateStatement);
```
Comment (ruxandraRosu) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java]: same as above, executeUpdate is the right statement

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java [hunk 9]
```diff
@@ -0,0 +1,463 @@
...
+        if (queryStatus){
+            queryStatus = executeCustomizedQuery(setBusinessTableDataSource);
```
Comment (ruxandraRosu) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java]: same as above, executeUpdate is the right statement

File: _layouts/default.html [hunk 10]
```diff
@@ -1,47 +0,0 @@
-<!-- SPDX-License-Identifier: CC-BY-4.0 -->
```
Comment (planetf1) [inline _layouts/default.html]: I noticed these files were out of sync with .gitignore in https://github.com/odpi/egeria/issues/773 - however I wasn't sure of the impact of deletion since they appear to relate to the website. Have you been able to confirm the effect & check with the owner? If it's known good perhaps worth adding a comment to that issue too

File: Community-Guide.md [hunk 11]
```diff
@@ -1,267 +1,267 @@
-<!-- SPDX-License-Identifier: CC-BY-4.0 -->
```
Comment (planetf1) [inline Community-Guide.md]: Is there a reason we have a huge change to this file?

File: open-metadata-implementation/adapters/open-connectors/data-store-connectors/structured-file-connector/src/test/resources/ComplexColumnsWithColumnNames.csv [hunk 12]
```diff
@@ -1,27 +1,27 @@
-RecId,ContactType,FirstName,LastName,Company,JobTitle,WorkLocation
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/data-store-connectors/structured-file-connector/src/test/resources/ComplexColumnsWithColumnNames.csv]: another global file change? Formatting? necessary?

File: open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventConsumer.java [hunk 13]
```diff
@@ -1,315 +1,315 @@
-/* SPDX-License-Identifier: Apache-2.0 */
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventConsumer.java]: what is the change here? SHould it be in this commit at all?

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/pom.xml [hunk 14]
```diff
@@ -24,5 +24,30 @@
...
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>repository-services-apis</artifactId>
+            <version>${open-metadata.version}</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.derby</groupId>
+            <artifactId>derbyclient</artifactId>
+            <version>10.14.1.0</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/pom.xml]: The deprecated virtualizer used a variable derby.version
The gaian connector is using 10.11.1.1
10.14 is much newer, but I have previously noticed some compatability issues. I would be inclined to be consistent and go with the version shipped by gaian - though if tested and happy that's ok.
If so, worth modifying the other versions & moving this hard version number to an appropriate property

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/auditlog/DerbyConnectorAuditCode.java [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+    CONNERCTOR_LOGICAL_TABLE_ERROR("DERBYCONNECTOR-0006.",
+            OMRSAuditLogRecordSeverity.EXCEPTION,
+            "The name of the logical table is not valid ",
+            "The local server is unable to create a derby connector.",
+            "Check if the name of the logical is valid"),
+    CONNERCTOR_SERVER_CONNECTION_ERROR("DERBYCONNECTOR-0007.",
+            OMRSAuditLogRecordSeverity.EXCEPTION,
+            "The address of the virtualization solution cannot be connected.",
+            "The local server is unable to create a derby connector.",
+            "Check id the address of the solution is accessible."),
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/auditlog/DerbyConnectorAuditCode.java]: if

Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/auditlog/DerbyConnectorAuditCode.java]: what's the address of the solution?

File: open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-api/pom.xml [hunk 16]
```diff
@@ -14,20 +14,26 @@
...
         <dependency>
             <groupId>org.odpi.egeria</groupId>
             <artifactId>repository-services-apis</artifactId>
             <version>${open-metadata.version}</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>view-generator-connectors</artifactId>
+            <version>1.1-SNAPSHOT</version>
```
Comment (planetf1) [inline open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-api/pom.xml]: change to ${open-metadata-version}

File: open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-api/pom.xml [hunk 17]
```diff
@@ -14,20 +14,26 @@
...
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>information-view-api</artifactId>
+            <version>1.1-SNAPSHOT</version>
```
Comment (planetf1) [inline open-metadata-implementation/governance-servers/virtualization-services/virtualization-services-api/pom.xml]: change to ${open-metadata-version}

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/CompDir-ContactEmail.csv [hunk 18]
```diff
@@ -1,35 +1,35 @@
-RecId;EType;Email
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/CompDir-ContactEmail.csv]: why this change?

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/CompDir-ContactList.csv [hunk 19]
```diff
@@ -1,27 +1,27 @@
-RecId;ContactType;FirstName;LastName;Company;JobTitle;WorkLocation
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/CompDir-ContactList.csv]: why this change

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/CompDir-ContactPhone.csv [hunk 20]
```diff
@@ -1,29 +1,29 @@
-RecId;ContactType;Number
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/CompDir-ContactPhone.csv]: do we need this change

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/EmplSAnl-EmpSalaryAnalysis.csv [hunk 21]
```diff
@@ -1,20 +1,20 @@
-HDR;RECTYPE;SERVICE;EMPSTATUS;FNAME;LNAME;PNUM;DEPT;LVL;LOCATION;LOCCODE;HOL;ROLE;ETYPE;SALARY;BONUS;SNUM;STREET;CITY;STATE;TAX;TAXP
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/EmplSAnl-EmpSalaryAnalysis.csv]: this one?

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/Employee-Dept.csv [hunk 22]
```diff
@@ -1,10 +1,10 @@
-DEPCODE;DEPNAME;MANAGER
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/Employee-Dept.csv]: do we need this change?

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/Employee-Employee.csv [hunk 23]
```diff
@@ -1,20 +1,20 @@
-PNUM;FNAME;LNAME;EMPSTATUS;LVL;DEPT;ROLE;LOCCODE
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/Employee-Employee.csv]: do we need this change - looks like accidental inclusion

File: open-metadata-resources/open-metadata-samples/access-services-samples/asset-management-samples/ContactList.csv [hunk 24]
```diff
@@ -1,27 +1,27 @@
-﻿RecId,ContactType,FirstName,LastName,Company,JobTitle,WorkLocation
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-samples/access-services-samples/asset-management-samples/ContactList.csv]: accidental change?

File: open-metadata-resources/open-metadata-samples/access-services-samples/asset-management-samples/CompanyDirectoryDatabase.csv [hunk 25]
```diff
@@ -1,29 +1,29 @@
-﻿RecId,ContactType,Number
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-samples/access-services-samples/asset-management-samples/CompanyDirectoryDatabase.csv]: accidental change?

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/ldap/03-groups.ldif [hunk 26]
```diff
@@ -1,73 +1,73 @@
-# SPDX-License-Identifier: Apache-2.0
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/ldap/03-groups.ldif]: accidental change?

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/Patient-Patient.csv [hunk 27]
```diff
@@ -1,33 +1,33 @@
-PatientId;Patient;LH Addr 1;LH Addr 2;LH Addr 3;LH Addr 4;LH Addr 5;Trial
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/Patient-Patient.csv]: accidental change?

File: open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/Location-WorkLocation.csv [hunk 28]
```diff
@@ -1,5 +1,5 @@
-WLID;WLName;;;;;;;
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/sample-data/coco-pharmaceuticals/data-files/Location-WorkLocation.csv]: accidental change?
